scrowl-io/scrowl
# [sc-644] - Chore/Update Lint Configs

### Short summary
Review and update linters configuration:

- I moved some configs from the `.vscode/settings.json` to the proper linters, this way we will not have VS Code handling any kind of code formatting and the installed tools will take care of it;
- Prettier should be the main code formatting tool, so it will be the last to run in the linting stack;
- I've added config files to each project, this way each one can have their specific rules on top of the global ones.

---

### Test steps

1. In the root folder run `yarn lint:check:course` to check the possible issues flagged by Eslint, Stylelint, and Prettier. You may notice that Prettier exist with code: 1 error, it seems like something went wrong, but this is the default behaviour when the Prettier validation fails. Running the next command will fix all of them.
2. Still in the root, run `yarn lint:fix:course`. You can use the VS Code Source Control (^ + shift + G) to check out the changes. ESLint isn't able to fix all issues automatically, like this one:
`apps/course-authoring/src/main/index.ts:26:3 - Error - Require statement not part of import statement.`
But it will be fixed in a different story.

--- 

### Real Time Validation

You can have real-time warnings/errors from Eslint, Stylelint, and Prettier using the VS Code Extensions from [this document](https://www.notion.so/Visual-Studio-Code-Extensions-d030c07591c140ae970457a84dd61cdf):

![image](https://user-images.githubusercontent.com/98419841/163206679-e5be316d-4dcd-4860-8c17-cf1627c9d4e0.png)
